x-pack/osquerybeat/ext/osquery-extension/pkg/filters/filters.go

```diff
@@ -0,0 +1,219 @@
+		return cast.ToUint64(field) < cast.ToUint64(f.Expression)
+	case reflect.Float64, reflect.Float32:
+		return cast.ToFloat64(field) < cast.ToFloat64(f.Expression)
+	case reflect.Bool:
+		return cast.ToInt8(field.(bool)) < cast.ToInt8(cast.ToBool(f.Expression))
+	default:
+		return false
+	}
+}
+
+// greaterThan checks if the value of the field is greater than the expression
+func (f Filter) greaterThan(entry any) bool {
+	field, kind, err := GetValueByOsqueryTag(entry, f.ColumnName)
+	if err != nil {
+		return false
+	}
+	switch kind {
+	case reflect.String:
+		castedField, err := cast.ToInt64E(field.(string))
+		if err != nil {
+			return false
+		}
+		castedExpression, err := cast.ToInt64E(f.Expression)
+		if err != nil {
+			return false
+		}
+		return castedField > castedExpression
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		return cast.ToInt64(field) > cast.ToInt64(f.Expression)
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		return cast.ToUint64(field) > cast.ToUint64(f.Expression)
+	case reflect.Float64, reflect.Float32:
+		return cast.ToFloat64(field) > cast.ToFloat64(f.Expression)
+	case reflect.Bool:
+		return cast.ToInt8(field.(bool)) > cast.ToInt8(cast.ToBool(f.Expression))
+	default:
+		return false
+	}
+}
+
+// like checks if the value of the field is like the expression
+func (f Filter) like(entry any) bool {
+	field, kind, err := GetValueByOsqueryTag(entry, f.ColumnName)
+	if err != nil {
+		return false
+	}
+	if kind != reflect.String {
+		return false
+	}
+	pattern := strings.ReplaceAll(f.Expression, "%", "*")
+	pattern = strings.ReplaceAll(pattern, "_", ".")
+	pattern = "^" + pattern + "$"
+	matched, err := regexp.MatchString(pattern, field.(string))
+	if err != nil {
+		return false
+	}
+	return matched
+}
+
+// glob checks if the value of the field is a glob pattern
+func (f Filter) glob(entry any) bool {
+	field, kind, err := GetValueByOsqueryTag(entry, f.ColumnName)
+	if err != nil {
+		return false
+	}
+	if kind != reflect.String {
+		return false
+	}
+	pattern := strings.ReplaceAll(f.Expression, "*", ".*")
+	pattern = strings.ReplaceAll(pattern, "?", ".")
+	pattern = "^" + pattern + "$"
+	matched, err := regexp.MatchString(pattern, field.(string))
+	if err != nil {
+		return false
+	}
+	return matched
+}
+
+// Matches checks if the value of the field matches the expression
+func (f Filter) Matches(entry any) bool {
+	switch f.Operator {
+	case table.OperatorEquals:
+		return f.equals(entry)
+	case table.OperatorGreaterThan:
+		return f.greaterThan(entry)
+	case table.OperatorLessThan:
+		return f.lessThan(entry)
+	case table.OperatorGreaterThanOrEquals:
+		return f.equals(entry) || f.greaterThan(entry)
+	case table.OperatorLessThanOrEquals:
+		return f.equals(entry) || f.lessThan(entry)
+	case table.OperatorLike:
+		return f.like(entry)
+	case table.OperatorGlob:
```

Comment:
are we not setting Regexp? If we end up doing it might be a good idea to store a compiled version of it